src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -576,252 +576,264 @@ private BoundExpression BindExpressionInternal(ExpressionSyntax node, BindingDia
                 return BadExpression(node, LookupResultKind.NotAValue);
             }
 
-            Debug.Assert(node != null);
-            switch (node.Kind())
+            BoundExpression result = bindExpressionInternal(node, diagnostics, invoked, indexed);
+
+            if (IsEarlyAttributeBinder && result.Kind == BoundKind.MethodGroup && (!IsInsideNameof || EnclosingNameofArgument != node))
```

Comment:
I was wondering in what scenario is `EnclosingNameofArgument != node` needed for correct behavior?

Reply:
> I was wondering in what scenario is `EnclosingNameofArgument != node` needed for correct behavior?

I have taken a conservatives approach to check for scenarios that I think can result in a successful binding in a most precise way. It is quite possible that I am missing some scenario and this condition is too strict, but I cannot think of any at the moment.